<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>‘Rewilding’ Hawaiian Picture-Wing Fly Microbiomes</title>
    <meta charset="utf-8" />
    <meta name="author" content="Kelli Konicek" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# ‘Rewilding’ Hawaiian Picture-Wing Fly Microbiomes
]
.author[
### Kelli Konicek
]
.institute[
### OCN682 Final Project
]
.date[
### 2024/12/10 (updated: 2024-12-10)
]

---






---
class: center, middle

# Hawaiian picture-wing flies

.pull-left[![_Drosophila hemipeza_](photos/h_natgeo.jpg)] 
.pull-right[![_Drosophila montgomeryi_](photos/m_natgeo.jpg)]



---
class: center, middle

# Picture-wings in the lab and the wild
.pull-left[![](photos/I_love_my_baby_flies.gif)]
.pull-right[![](photos/wild.jpeg)]

---
class: center, middle

![](photos/microbiome.png)


---
class: center, middle
![](photos/setup.png)
---
class: center, middle
# Let's analyze microbiome data 
## Load Libraries

``` r
library(phyloseq)
library(here)
library(BiocManager)
library(tidyverse)
library(microbiome)
library(scales)
library(data.table)
library(patchwork)
library(janitor)
```

---
class: center, middle
# Data sets you need for microbiome analysis
## 1. Operational Taxonomic Unit (OTU)  matrix

.pull-center[![](photos/otu.png)]

---
class: center, middle
# Data sets you need for microbiome analysis
## 2. Taxonomy assigned to OTU numbers

.pull-center[![](photos/taxa.png)]

---
class:middle, center
# Data sets you need for microbiome analysis
## 3. Metadata
.pull-center[![](photos/meta.png)]

---

# Upload ITS (Fungal Amplicon) Data into a Phyloseq Object 


``` r
ps &lt;- read_mothur2phyloseq((here("data","abundance_table-lulu-100.shared")),here("data", "taxonomy_100.taxonomy"), mapping.file = (here("data","ITS_meta.csv"))) #upload the three files needed

ps_rewild &lt;- subset_samples(ps, species == "Drosophila_montgomeryi")

# Extract individual files to double-check things worked

OTU = as(otu_table(ps_rewild), "matrix") # OTU table extract (rows are asvs)

taxonomy &lt;- data.frame(tax_table(ps_rewild)) #tax table extract 

metadata &lt;- data.frame(sample_data(ps_rewild)) #metadata extract
```

---
class:middle, center
# Sequencing depth histogram- how did my samples sequence?


``` r
depth &lt;- metadata %&gt;%
  mutate(TotalReads = sample_sums(ps_rewild), keep.rownames = TRUE) %&gt;%
  ggplot(aes(TotalReads, fill = treatment)) +
  geom_histogram(bins = 30) +
  scale_x_continuous(breaks=seq(0,25000,1500)) +
  scale_y_continuous(breaks=seq(0,2,1)) +
  theme(axis.text.x.bottom = element_text(size = 10, angle = 60, hjust = .7, vjust = .7)) +
  scale_fill_discrete(labels=c('Lab Only Flies', 'Rewilded Flies')) +
  labs(title = "Sequencing Depth",
       x = "Number of reads per sample",
       y = "Samples with x number of reads",
       fill = "Sample Type") 
depth
```

![](../output/depth-1.png)&lt;!-- --&gt;
---

## Sequencing depth histogram- how did my samples sequence?

``` r
depth
```

![](konicek_final_slides_files/figure-html/fig1-1.png)&lt;!-- --&gt;


---

# Relative Abundance Bar Plot from a Phyloseq Object
merge asvs to shared Genus, and get relative abundance of top 11 taxa (10 known plus unknown genera)

``` r
psGLOM &lt;- tax_glom(ps_rewild, taxrank = "Genus") #merge taxa to Genus level
top11G.names = sort(tapply(taxa_sums(psGLOM), tax_table(psGLOM)[, "Genus"], sum), TRUE)[1:11] #make a list of top 11 names
top11G = subset_taxa(psGLOM, Genus %in% names(top11G.names)) # filter phyloseq object by the list of names
ps0 &lt;- transform_sample_counts(top11G, function(x) x / sum(x)) #transform abundance counts into relative abundance
df &lt;- psmelt(ps0) #turn the phyloseq object into a dataframe
df$Genus &lt;- gsub("NA", "Unknown Genera", df$Genus) #Let's say "Unknown Genera" instead of NA.
```

---
# Relative Abundance Bar Plot from a Phyloseq Object

``` r
plotm &lt;- df %&gt;%
  ggplot() +
  geom_bar(aes(x = Sample,
               y = Abundance,
               color=Genus,
               fill=Genus),
           stat="identity",
           position="stack") +
  scale_y_continuous(labels = percent) +
  theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        strip.background = element_blank()) +
  labs(title = "Microbiome Profiles: Lab Reared vs. Rewilded",
       x = "Fly microbiome profiles",
       y = "Relative abundance of taxa") +
  facet_grid(~treatment,labeller = as_labeller(c(`Lab_Only` = "Lab Reared Flies", `rewilded` = "Rewilded Flies")))
plotm
```

![](../output/relabnd-1.png)&lt;!-- --&gt;


---
Relative Abundance Bar Plot from a Phyloseq Object

``` r
plotm
```

![](konicek_final_slides_files/figure-html/plotm-1.png)&lt;!-- --&gt;


---
# Final Figure: Beta Diversity using PCoA ordination

``` r
dist = phyloseq::distance(ps_rewild, method="bray", binary = TRUE) #Measure dissimilarity between samples
ordination = ordinate(ps_rewild, method="PCoA", distance=dist) # ordinate those dissimilarity measurements onto a Principle Coordinate Analysis Plot
beta &lt;- plot_ordination(ps_rewild, ordination, color ="treatment") + #plot_ordination starts the plot
  theme_classic() +
  theme(strip.background = element_blank()) +
  geom_point() +
  stat_ellipse(aes(group = treatment), level = 0.95, linetype = 2) +
  scale_color_discrete(breaks = c("Lab_Only", "rewilded"), name = "Sample Type", labels = c("Lab Reared Flies", "Rewilded Flies")) +
  labs(title = "Fly microbiome composition differs by sample type",
       caption = "PCoA using Bray-Curtis dissimiliarity")
beta
```

![](../output/beta1-1.png)&lt;!-- --&gt;


---


``` r
beta
```

![](konicek_final_slides_files/figure-html/beta-1.png)&lt;!-- --&gt;

---
# Conclusions
Flies that stayed in the lab sure have different microbiomes compared to flies that lived in the wild! 

# Caveats 
These stats are Real Bad and we have learned nothing quantifiable, but I learned how to get the figures up and running! 
---

class: center, middle

# Mahalo!

Slides created via the R package [**xaringan**](https://github.com/yihui/xaringan).

    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
